autorest/adal/token.go

```diff
@@ -655,9 +657,10 @@ func newServicePrincipalTokenFromMSI(msiEndpoint, resource string, userAssignedI
 			AutoRefresh:   true,
 			RefreshWithin: defaultRefresh,
 		},
-		refreshLock:      &sync.RWMutex{},
-		sender:           &http.Client{},
-		refreshCallbacks: callbacks,
+		refreshLock:           &sync.RWMutex{},
+		sender:                &http.Client{},
+		refreshCallbacks:      callbacks,
+		MaxMSIRefreshAttempts: 5,
```

Comment:
can we make this a constant like `defaultRefresh` ?

Reply:
also this way if you want you can remove the comment that is 5 by default and just say the default is the var name so we dont risk of going out of sync